src/box/txn.c

```diff
@@ -1246,9 +1245,8 @@ txn_on_stop(struct trigger *trigger, void *event)
 static int
 txn_on_yield(struct trigger *trigger, void *event)
 {
-	(void) trigger;
 	(void) event;
-	struct txn *txn = in_txn();
+	struct txn *txn = (struct txn *) trigger->data;
```

Comment:
Please, extract these changes into a separate commit. Also we do not use whitespace after type cast (in the new code). Just like with all unary operators.